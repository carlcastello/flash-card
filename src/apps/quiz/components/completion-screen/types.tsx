export interface IOwnProps {
  onClose: () => void,
  onReview: () => void,
  classes: {
    
  }
}