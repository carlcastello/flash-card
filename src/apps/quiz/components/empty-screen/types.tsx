export interface IOwnProps {
  onEdit: () => void,
  onClose: () => void,
  classes: {

  }
}