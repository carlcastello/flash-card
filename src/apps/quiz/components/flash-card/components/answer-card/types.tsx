import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  answer: string,
  flashcardStatus: FlashcardStatus,
  ref: any,
  classes: {
    paper: string,
    input: string,
    typography: string
  }
}

export interface IOwnState {
  answer: string,
  hideAnswer: boolean,
}