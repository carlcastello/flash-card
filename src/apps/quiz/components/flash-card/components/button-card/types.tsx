import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  flashcardStatus: FlashcardStatus,
  submit: () => void,
  next: () => void,
  classes: {
    button: string
  }
}

export interface IOwnState {

}