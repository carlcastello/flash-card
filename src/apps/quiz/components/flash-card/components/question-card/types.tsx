import { FlashCardStatus } from "../../types";
import { QuestionType } from "../../../../../commons/types";

export interface IQuestionObject {
  question: string,
  subQuestion?: string,
  hint?: string,
  questionType: QuestionType,
}

export interface IOwnProps {
  flashCardStatus: FlashCardStatus,
  questionObject: IQuestionObject,
  onSkip: () => void,
  classes: {
    paper: string,
    boxButtonContainer: string,
    boxQuestionaireContainer: string
  }
}