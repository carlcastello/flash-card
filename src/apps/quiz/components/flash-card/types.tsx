import { IFlashCard } from "../../../commons/types";

export enum FlashCardStatus {
  DEFAULT = "DEFAULT",
  CORRECT = "CORRECT",
  WRONG = "WRONG",
  HINT = "HINT"
}

export interface IOwnProps {
  update: (flashCardStatus: FlashCardStatus) => void,
  next: () => void,
  flashCardObject: IFlashCard,
  classes: {

  }
}

export interface IOwnState {
  flashCardStatus: FlashCardStatus
}