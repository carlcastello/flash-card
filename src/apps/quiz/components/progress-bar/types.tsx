import { FlashcardStatus } from "../flash-card/types";

export interface IOwnProps {
  currentQuestion: number,
  totalQuestion: number,
  flashcardStatus: FlashcardStatus,
  classes: {
    title: string,
    progressBarContainer: string,
    progressBar: string,
  }
}