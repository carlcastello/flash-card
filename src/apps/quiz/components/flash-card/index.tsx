import React, { Component, ReactNode, createRef } from 'react';

import { withStyles, Box } from '@material-ui/core';

import QuestionCard from './components/question-card';
import AnswerCard from './components/answer-card';
import ButtonCard from './components/button-card';

import { IOwnProps, FlashcardStatus } from './types';
import styles from './styles';


export class Flashcard extends Component<IOwnProps> {

  answerCardRef: any = createRef();   

  onSubmit = (): void => {
    const answer = this.answerCardRef.current.state.answer;

    const {
      question: {
        answer: correctAnswer
      },
      update,
    } = this.props;

    var flashcardStatus: FlashcardStatus = FlashcardStatus.DEFAULT;

    if (answer === correctAnswer) {
      flashcardStatus = FlashcardStatus.CORRECT;
    } else {
      flashcardStatus = FlashcardStatus.WRONG;
    }

    update(flashcardStatus);
  }


  onNext = (): void => {
    const {
      next,
    } = this.props;

    next();
    this.answerCardRef.current.setState({answer: ''});
  }

  onSkip = (): void => {
    const {
      update,
    } = this.props;

    update(FlashcardStatus.HINT)
  }

  renderQuestionCard(): ReactNode {
    const {
      props: {
        flashcardStatus,
        question: {
          question,
          subQuestion,
          hint
        }
      }
    } = this;
    return (
      <QuestionCard
        flashcardStatus={flashcardStatus}
        onSkip={this.onSkip}
        questionObject={{
          question,
          subQuestion,
          hint
        }}/>
    );
  }

  renderAnswerCard(): ReactNode {
    const {
      props: {
        question: {
          answer,
        }
      },
      props: {
        flashcardStatus
      }
    } = this;
    return (
      <AnswerCard 
        answer={answer}
        flashcardStatus={flashcardStatus}
        ref={this.answerCardRef}/>
    );
  }

  renderButtonCard(): ReactNode {
    const {
      onSubmit,
      onNext,
      props: {
        flashcardStatus
      },
    } = this;
    return (
      <ButtonCard 
        flashcardStatus={flashcardStatus}
        next={onNext}
        submit={onSubmit}/>
    )
  }

  render(): ReactNode {
    return(
      <Box pt={2.5} px={5}>
        <form>
          {this.renderQuestionCard()}
          {this.renderAnswerCard()}
          {this.renderButtonCard()}
        </form>
      </Box>
    );
  }
}

export default withStyles(styles)(Flashcard);