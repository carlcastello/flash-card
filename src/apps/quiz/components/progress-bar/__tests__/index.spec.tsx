import React from "react";
import { shallow } from "enzyme";

import { ProgressBar } from "../index";
import { FlashcardStatus } from "../../flash-card/types";
import toJson from "enzyme-to-json";
import { LinearProgress } from "@material-ui/core";

describe("", () => {
  
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <ProgressBar
        currentQuestion={3}
        totalQuestion={4}
        flashcardStatus={FlashcardStatus.CORRECT}
        classes={{
          title: "tittle",
          progressBarContainer: "progress-bar-container",
          progressBar: "progress-bar"
        }}/>
    )
  });

  describe("the render function", () => {
    it("renders", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(LinearProgress)).toHaveLength(1);
      expect(wrapper.find(LinearProgress).props().value).toEqual(75);
    });
  });
});