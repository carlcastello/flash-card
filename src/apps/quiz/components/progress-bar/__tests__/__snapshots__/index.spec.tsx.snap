// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` the render function renders 1`] = `
<Styled(MuiBox)
  className="progress-bar-container"
>
  <WithStyles(ForwardRef(Typography))
    className="tittle"
    variant="subtitle1"
  >
    Question 
    3
     of 
    4
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(LinearProgress))
    className="progress-bar CORRECT"
    value={75}
    variant="determinate"
  />
</Styled(MuiBox)>
`;
